Modules/Loadable/Markups/MRML/vtkMRMLMarkupsCurveNode.cxx

```diff
@@ -183,6 +194,182 @@ double vtkMRMLMarkupsCurveNode::GetCurveLengthBetweenStartEndPointsWorld(vtkIdTy
   }
 }
 
+//---------------------------------------------------------------------------
+void vtkMRMLMarkupsCurveNode::ResampleCurveSurface(double controlPointDistance, vtkMRMLModelNode* modelNode, double maximumSearchRadius)
+{
+	if(!modelNode)
+	  {
+	  vtkErrorMacro("vtkMRMLMarkupsCurveNode::ResampleCurveSurface failed: Constraint surface is not valid");
+	  }
+	vtkPoints* originalPoints = this->GetCurvePointsWorld();
+	if (!originalPoints  || originalPoints->GetNumberOfPoints() < 2)
+      {
```

Comment:
indentation

Reply:
Don't we need to still snap the control point to the surface?

Reply:
Since the assumption is that the control points are already snapped to the surface, in the case of a curve with a single point, I think it may make more sense to exit. I added an error message for this case.

Reply:
It's a pity that we cannot snap a single point, logging an error message is an acceptable limitation for now.